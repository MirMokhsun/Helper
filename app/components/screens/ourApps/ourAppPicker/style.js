import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
    container: {
        width: 'auto',
        height: 90,
        borderBottomWidth: 1,
        borderBottomColor: '#D3D9E7',
        marginHorizontal: 15,
        flexDirection: 'row',
        alignItems: 'center',
        padding: 8,
    },
    text: {
        fontFamily: 'Roboto-Light',
        fontSize: 16,
        color: '#000756',
        lineHeight: 65,
        marginLeft: 24,
    },
    image: {
        width: 56,
        height: 56,
        borderRadius: 5,
    }
});