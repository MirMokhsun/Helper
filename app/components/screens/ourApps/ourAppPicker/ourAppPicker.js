import React from 'react';
import { Text, TouchableOpacity, Image, Linking, Platform } from 'react-native';
import { styles } from './style';
import AppLink from 'react-native-app-link';

export class OurAppPicker extends React.PureComponent {
    constructor(props) {
        super(props);
    }

    goToStore = () => {
        const {
            playStoreId,
            appStorelink
        } = this.props;

        if(Platform.OS === 'ios'){
            Linking.openURL(appStorelink);
        } else {
            AppLink.openInStore({
                playStoreId
            }).then(() => {
                console.warn('opened');
            })
        }
    }

    render() {
        const { text, image, id } = this.props;
        return (
            <TouchableOpacity style={styles.container} onPress={this.goToStore} testID = {`ourAppPicker${id}ID`} accessibilityLabel={`ourAppPicker${id}ID`}>
                {image ? <Image source={image} style={styles.image} /> : null}
                <Text style={styles.text}>
                    {text}
                </Text>
            </TouchableOpacity>
        );
    }
}   
