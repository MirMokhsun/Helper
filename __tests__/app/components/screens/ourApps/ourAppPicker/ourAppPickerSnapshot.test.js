import 'react-native';
import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { OurAppPicker } from '../../../../../../app/components/screens/ourApps/ourAppPicker/ourAppPicker';

it('create snapshot OurAppPicker', () => {
    const renderer = new ShallowRenderer()
    let ourAppPicker = renderer.render(<OurAppPicker />);
    expect(ourAppPicker).toMatchSnapshot();
});