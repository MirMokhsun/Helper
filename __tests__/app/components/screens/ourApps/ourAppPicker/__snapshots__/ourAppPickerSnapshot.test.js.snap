// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create snapshot OurAppPicker 1`] = `
<TouchableOpacity
  accessibilityLabel="ourAppPickerundefinedID"
  activeOpacity={0.2}
  onPress={[Function]}
  style={
    Object {
      "alignItems": "center",
      "borderBottomColor": "#D3D9E7",
      "borderBottomWidth": 1,
      "flexDirection": "row",
      "height": 90,
      "marginHorizontal": 15,
      "padding": 8,
      "width": "auto",
    }
  }
  testID="ourAppPickerundefinedID"
>
  <Text
    style={
      Object {
        "color": "#000756",
        "fontFamily": "Roboto-Light",
        "fontSize": 16,
        "lineHeight": 65,
        "marginLeft": 24,
      }
    }
  />
</TouchableOpacity>
`;
